### PR TITLE
Hire stream section limits

### LINKED ISSUE #1006: Hire stream section limits
1. Add hire stream settings page
2. Allow interview limits per day and per week
3. Filter interviewers weekly for list interviews and daily + weekly for schedules

### PR DESCRIPTION
**Pull request by tsumo**
## PR includes

-   [x] Feature

## Related issues

Resolve #1006

### CODE REVIEW

File: src/components/SlotCalendarEvent/SlotCalendarEvent.tsx [hunk 1]
```diff
@@ -19,16 +20,29 @@ export const SlotCalendarEvent: ComponentType<SlotCalendarEventProps> = ({ event
...
-        <div>
-            <Text size="xs" className={s.CreatorName}>
-                {creator?.name}
-            </Text>
-
-            <Text color="textSecondary" size="xs">
-                {title}
-            </Text>
+        <Text size="xs">
+            <Text>{creator?.name}</Text>
```
Comment (LamaEats) [inline src/components/SlotCalendarEvent/SlotCalendarEvent.tsx:24]: Really needs text component inside the text component? 🙃

File: src/components/SlotCalendarEvent/SlotCalendarEvent.tsx [hunk 2]
```diff
@@ -19,16 +20,29 @@ export const SlotCalendarEvent: ComponentType<SlotCalendarEventProps> = ({ event
...
+
+            <Text>{title}</Text>
+
             <CalendarEventLinkedSection interviewSection={interviewSection} sectionTitleOnly />
-        </div>
+
+            {nullable(event.unavailableDueToWeekLimit, () => (
+                <Alert
```
Comment (LamaEats) [inline src/components/SlotCalendarEvent/SlotCalendarEvent.tsx:31]: Also this two optional alert components may be placing outside of the text component

Comment (tsumo) [reply src/components/SlotCalendarEvent/SlotCalendarEvent.tsx:33]: Fixed

File: src/components/SlotCalendarEvent/SlotCalendarEvent.tsx (hunk 1 above)
Comment (tsumo) [reply src/components/SlotCalendarEvent/SlotCalendarEvent.tsx]: If I use <p> I would have to reset default paragraph margins

File: src/modules/calendarTypes.ts [hunk 3]
```diff
@@ -147,6 +148,14 @@ export interface CalendarEventInstance {
...
+    unavailableDueToDayLimit?: boolean;
+
+    unavailableDueToWeekLimit?: boolean;
 }
 
 export type CalendarData = CalendarEventInstance[];
+
+export type UnavailableUsersForWholeWeek = Set<number>;
+
+export type UnavailableUsersByWeekDay = Record<number, Set<number>>;
```
[the remark this answers is not in the document]
Comment (tsumo) [reply src/modules/calendarTypes.ts:161]: Just didn't want to type Record<Set...> three times in three different files

**Review by LamaEats — APPROVED** [review]

**Pull request merged**